server_functions/Filtering.R

```diff
@@ -0,0 +1,49 @@
+# function to take the user arguments (currently given in config) and filter the csv data based on these.
+# args: dat - the csv data file for a single indicator
+# returns a filtered tibble
+
+filter_for_selections <- function(dat){
+  
+  irrelevant_variables <- c("Observation.status", "Unit.multiplier", "Unit.measure", "GeoCode", "Value", "Year")
+  
+  selected_characteristics <- c(char1, char2, char3, char4)
+  
+  unselected_characteristics <- setdiff(names(dat), selected_characteristics)
+  relevant_unselected_characteristics <- setdiff(unselected_characteristics, irrelevant_variables)
+  
+  unselected_charateristics_removed <- dat %>%
```

Comment:
BUG: If one of the selected disaggregations is nested in an unselected disag, it will get dropped at this stage. e.g. in 8-5-1 age is nested under occupation, so removing all but the NAs in occupation leaves only NAs in age. I will look into a fix for this.

Reply:
I have started sorting out filtering when one column is entirely dependent on another.   
  
I am still facing issues, but I think this may be because I have made some impossible selections in the config file (mock up of user selections)

To do (me) - change config to only have feasible selections

a4e73c7baf4a034297d85e23e5391936cdb888d3

Reply:
Fix completed (but only tested for a single case)
a25fc2a4e37138517878d85abcfed2bcfad7a50e

Reply:
While using the original code as a base, I made quite a lot of edits to both the code and the structure on the server_functions_review branch so going to close this merge request and open a new one for @varunjakki and @morrid9 to review